aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -375,14 +375,56 @@ public void onError(Exception error) {
 
     @Override
     public void signIn(
-            @Nullable String username,
-            @Nullable String password,
-            @NonNull final Consumer<AuthSignInResult> onSuccess,
-            @NonNull final Consumer<AuthException> onException
+        @Nullable String username,
+        @NonNull AuthSignInOptions options,
+        @NonNull final Consumer<AuthSignInResult> onSuccess,
+        @NonNull final Consumer<AuthException> onException
+    ) {
+        Map<String, String> metadata = null;
+
+        if (options != null && options instanceof AWSCognitoAuthSignInOptions) {
+            metadata = ((AWSCognitoAuthSignInOptions) options).getMetadata();
+        }
+
+        awsMobileClient.signIn(username, metadata, new Callback<SignInResult>() {
```

Comment:
According to the build log, this line is failing to compile with this error: 
```
incompatible types: Map<String,String> cannot be converted to String
        awsMobileClient.signIn(username, metadata, new Callback<SignInResult>() {
```

Reply:
> According to the build log, this line is failing to compile with this error:
> 
> ```
> incompatible types: Map<String,String> cannot be converted to String
>         awsMobileClient.signIn(username, metadata, new Callback<SignInResult>() {
> ```

This is dependent on https://github.com/aws-amplify/aws-sdk-android. 
Actually, the underlying aws sdk(https://github.com/aws-amplify/aws-sdk-android) also has the same. It also assumes password by default. 
I also created a PR to add the APIs without password in the underlying SDK. Here is the PR https://github.com/aws-amplify/aws-sdk-android. 
This issue will be automatically resolved after the aws-sdk-android merges the PR mentioned above.